import eoapi-cdk constructs and reuse them

emileten opened the pull request:
Another take at https://github.com/developmentseed/eoAPI/pull/135. This one is much simpler. All modifications are in `infrastructure/aws`, and are all about swapping direct native AWS CDK calls with eoapi-cdk construct calls.

**_There is no change in behavior in this PR_** ; i.e, the deployed infrastructure should be identical for a given set of environment variables after this PR is merged. 

The code in this branch synthesized successfully and I'm trying to deploy right now.

emileten commented on infrastructure/aws/requirements-cdk.txt:
```diff
@@ -1,9 +1,11 @@
-aws-cdk-lib==2.94.0
-aws_cdk-aws_apigatewayv2_alpha==2.94.0a0
-aws_cdk-aws_apigatewayv2_integrations_alpha==2.94.0a0
+aws-cdk-lib==2.99.1
```
necessary for compatibility with eoapi-cdk, unlikely to break anything.

emileten commented on infrastructure/aws/cdk/app.py:
```diff
@@ -181,140 +91,120 @@ def __init__(  # noqa: C901
+            
             db_secrets = {
-                "POSTGRES_HOST": setup_db.secret.secret_value_from_json(
+                "POSTGRES_HOST": pgstac_db.pgstac_secret.secret_value_from_json(
```
we should be able to do away entirely with these because the secret arn is passed as an env var to the lambda but I'd have to change the runtime to parse the secret there, and I want to avoid changing the runtime at all in this repo, so I left it as is.

same comment applies for all lambdas where we pass the postgres env vars.

emileten commented on infrastructure/aws/cdk/app.py:
```diff
@@ -325,65 +215,53 @@ def __init__(  # noqa: C901
-                eostac_function.add_environment(key=k, value=str(v))
-
+            env.update(db_secrets)
             # If raster is deployed we had the TITILER_ENDPOINT env to add the Proxy extension
```
vincent do you have a doc link for this proxy extension ? haven't read about it and i'm curious !

vincentsarago replied:
it's not really a proxy. But for our custom eoAPI runtime we add a titiler extension https://github.com/developmentseed/eoAPI/blob/main/runtime/eoapi/stac/eoapi/stac/extension.py which is used to add a `/collections/{collectionId}/items/{itemId}/tilejson.json` endpoint to the STAC service which will redirect to the raster tilejson endpoint. This is really custom for our custom runtime

emileten commented:
One significant difference with the other PR, in addition, is that I'm not adding the ingestor & browser, I'd like to keep things simple and do that later in another PR if possible.

emileten commented:
Looks like it mostly works, here are links of services [vector](https://2u72j50q8a.execute-api.us-east-1.amazonaws.com/), [stac](https://nyotlq3hzi.execute-api.us-east-1.amazonaws.com/), [raster](https://gfxg1p8g0c.execute-api.us-east-1.amazonaws.com/)

vincentsarago commented:
> One significant difference with the other PR, in addition, is that I'm not adding the ingestor & browser, I'd like to keep things simple and do that later in another PR if possible.

🙏 yeah we def want stac-browser and ingestor to be deployed but 👍 for doing it later :-)

emileten commented on infrastructure/aws/cdk/config.py:
```diff
@@ -42,7 +42,7 @@ class eoDBSettings(BaseSettings):
     context: bool = True
     mosaic_index: bool = True
-
+    allocated_storage: int = 20
```
@vincentsarago actually that's not going to work ; the idea is that we can merge this, run `cdk deploy` on the existing staging stack and no changes will happen. I see that the current DB storage is 100GiB (devseed account, us-east1-1, RDS instance id `eoapi-staging-eoapistagingpostgresdbf3d7e91a-cvyhomoxwlck`).

AWS is going to throw an error with 20 because you can't decrease an RDS storage allocation.

So I'll replace this with 100.

vincentsarago replied:
🤷 default to 20 is fine. We can just overwrite when we need to update our stack

emileten commented:
@vincentsarago this is ready to be merged. Important notes : 

- I deployed this to a stack named `eoapi-dev` in the devseed account. I swapped the domain mappings : `*.eoapi.dev` point to APIs in this stack (I copied the content of the DB over). 
- removed the NAT gateway (and made corresponding necessary networking changes we discussed on slack).
- added the STAC ingestor. I created a custom domain name for this with a record at `ingestor.eoapi.dev`, for now it's not working but we might need to wait -- otherwise it's available at the non custom URL that's shown in the stack.
- added the STAC browser, see this public endpoint : http://eoapi-dev-stac-browser.s3-website-us-east-1.amazonaws.com.

emileten commented:
When you're 👍 I can merge and later delete the `eoapi-staging` stack.

emileten commented:
Edit : `ingestor.eoapi.dev` works now

vincentsarago commented on infrastructure/aws/cdk/config.py:
```diff
@@ -56,7 +57,8 @@ class eoSTACSettings(BaseSettings):
     memory: int = 256
-
+    stac_browser_github_tag: None | str = "v3.1.0"  # if not none, will try to deploy this version of radiant earth stac browser
+    stac_api_custom_domain_name: None | str = "https://stac.eoapi.dev"
```
🤔 can we avoid setting a default to our stack here?

emileten replied:
I could remove the default so that we're always force to explicitly write it in the dotenv file, yes. Is that what you mean? 

However if you're question is: "can't the cdk use the dynamic api gateway url instead, when the stack gets synthesised ?" The answer is unfortunately no, because that url is resolved only at deployment (whereas we need it before, during synth)

vincentsarago replied:
but if `None` later in the application can we default to either the STAC API gateway url or a custom endpoint for STAC that user would have put?

vincentsarago replied:
`stac_browser_github_tag` can have a default 👍

emileten commented:
@vincentsarago 
Yes the /docs is broken in the custom domain name, there is a ticket for this in eoAPI-cdk. I'll try to fix that tomorrow. 

You can reach /docs with the api gateway url though (not on my computer right now, but that url is one of the outputs of the stack).

emileten commented:
@vincentsarago 

> 🤔 can we avoid setting a default to our stack here?

☝️ done

> the /docs is broken

☝️ fixed https://ingestor.eoapi.dev/docs

vincentsarago commented on infrastructure/aws/cdk/app.py:
```diff
@@ -242,10 +242,10 @@ def __init__(  # noqa: C901
-                    eostac_settings.stac_api_custom_domain_name is not None
-                ), "stac_api_custom_domain_name must be set if stac_browser_github_tag is not None. The browser deployment needs a resolved STAC API url at deployment time and so needs to rely on a predefined custom domain name."
+                    eostac_settings.stac_browser_github_tag is not None
+                ), "stac_browser_github_tag must be set if stac_api_custom_domain_name is not None."
```
@emileten can you explain again the relationship between the `custom domain name` and the `GitHub tag`?

emileten replied:
sorry, the way I wrote it, it's confusing. 

In `eoSTACSettings`, I defined two parameters for the purpose of the STAC browser : 

- the catalog URL, which I named `stac_api_custom_domain_name`, that the browser is going to show
- the version of the browser, `stac_browser_github_tag`

These two things should probably go in another configuration class named `eoStacBrowserSettings` for clarity. 

And the assertion just checks that a version is provided if a catalog url is provided to avoid unclear errors downstream. 

Does that make sense @vincentsarago ?

vincentsarago replied:
👍 thanks 

> These two things should probably go in another configuration class named eoStacBrowserSettings for clarity.

That would be lovely
